website/addons/owncloud/static/owncloudUserConfig.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+
+var ko = require('knockout');
+var $ = require('jquery');
+var Raven = require('raven-js');
+var bootbox = require('bootbox');
+require('js/osfToggleHeight');
+var OAuthAddonSettingsViewModel = require('js/addonSettings.js').OAuthAddonSettingsViewModel;
+var language = require('js/osfLanguage').Addons.owncloud;
+var osfHelpers = require('js/osfHelpers');
+var oop = require('js/oop');
+var ExternalAccount = require('js/addonSettings').ExternalAccount;
+var $modal = $('#ownCloudCredentialsModal');
+
+
+var ViewModel = oop.extend(OAuthAddonSettingsViewModel,{
+    constructor: function(url){
+        var self = this;
+        self.super.constructor('owncloud','ownCloud');
+
+        const otherString = 'Other (Please Specify)';
+
+        self.username = ko.observable();
+        self.password = ko.observable();
+        self.urls = ko.observable({});
+        self.hosts = ko.observableArray([]);
+        self.selectedHost = ko.observable();
+        self.customHost = ko.observable();
+        self.loaded = ko.observable(false);
+
+        self.host = ko.pureComputed(function() {
+            return self.useCustomHost() ? self.customHost() : self.selectedHost();
+        });
+
+        self.visibleHosts = ko.pureComputed(function() {
+            return self.hosts().concat([otherString]);
+        });
+
+        self.useCustomHost = ko.pureComputed(function() {
+            return self.selectedHost() === otherString;
+        });
+
+        self.showCredentialInput = ko.pureComputed(function() {
+            return Boolean(self.selectedHost());
+        });
+
+        $.ajax({
+            url: url,
+            type: 'GET',
+            dataType: 'json'
+        }).done(function (response) {
+            var data = response.result;
+            self.urls(data.urls);
+            self.hosts(data.hosts);
+            self.loaded(true);
+            self.updateAccounts();
+        }).fail(function (xhr, textStatus, error) {
+            self.setMessage(language.userSettingsError, 'text-danger');
+            Raven.captureMessage('Could not GET OwnCloud settings', {
+                url: url,
+                textStatus: textStatus,
+                error: error
+            });
+        });
+    },
+
+    clearModal : function() {
+        var self = this;
+        self.message('');
+        self.messageClass('text-info');
+        self.username(null);
+        self.password(null);
+        self.selectedHost(null);
+        self.customHost(null);
+    },
+
+    updateAccounts : function() {
```

Comment:
I think this might override be superfluous. It looks like the only thing it's doing differently from the parent is assigning fields that are on the output from `<Addon>Serializer.serialize_account`... effectively re-mapping a field that gets passed to the front-end already (`providerUrl`)